core/controllers/skill_editor.py

```diff
@@ -139,60 +137,41 @@ def get(self, skill_id):
         self.render_json(self.values)
 
 
-class SkillEditorQuestionHandler(base.BaseHandler):
-    """Manages the creation of a question and receiving of all question
-    summaries for display in skill editor page.
-    """
-    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
-
-    @acl_decorators.can_edit_skill
-    def get(self, skill_id):
-        """Handles GET requests."""
-        skill_domain.Skill.require_valid_skill_id(skill_id)
-
-        start_cursor = self.request.get('cursor')
-
-        question_summaries, skill_descriptions, next_start_cursor = (
-            question_services.get_question_summaries_and_skill_descriptions(
-                constants.NUM_QUESTIONS_PER_PAGE, [skill_id], start_cursor)
-        )
-        return_dicts = []
-        for index, summary in enumerate(question_summaries):
-            return_dicts.append({
-                'summary': summary.to_dict(),
-                'skill_description': skill_descriptions[index]
-            })
-
-        self.values.update({
-            'question_summary_dicts': return_dicts,
-            'next_start_cursor': next_start_cursor
-        })
-        self.render_json(self.values)
-
-
 class EditableSkillDataHandler(base.BaseHandler):
     """A data handler for skills which supports writing."""
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
-    @acl_decorators.can_edit_skill
-    def get(self, skill_id):
-        """Populates the data on the individual skill page."""
-        skill_domain.Skill.require_valid_skill_id(skill_id)
-        skill = skill_services.get_skill_by_id(skill_id, strict=False)
-        if skill is None:
-            raise self.PageNotFoundException(
-                Exception('The skill with the given id doesn\'t exist.'))
+    @acl_decorators.can_edit_skills
+    def get(self, skill_ids):
+        """Populates the data on the individual or multiple skill pages."""
+
+        try:
+            skill_ids = skill_ids.split(',')
+        except Exception:
+            raise self.PageNotFoundException
+
+        skill_dicts = []
+        for skill_id in skill_ids:
+            skill_domain.Skill.require_valid_skill_id(skill_id)
+            skill = skill_services.get_skill_by_id(
+                skill_id, strict=False)
+            if skill is None:
+                raise self.PageNotFoundException(
+                    Exception('The skill with the given id doesn\'t exist.'))
+            skill_dicts.append(skill.to_dict())
 
         self.values.update({
-            'skill': skill.to_dict()
+            'skills': skill_dicts
         })
 
         self.render_json(self.values)
 
-    @acl_decorators.can_edit_skill
-    def put(self, skill_id):
+    @acl_decorators.can_edit_skills
+    def put(self, skill_ids):
```

Comment:
This is a bit confusing. You are taking a list of skill_ids but updating only the first one? Either update all of them, or do not take in multiple skills to update (I prefer the latter)

Reply:
Since this is in the same handler as 'get', and 'get' takes in skill_ids as its argument , 'put' has to take in skill_ids instead of skill_id as well. Is there any alternative way?

Reply:
Wait I don't get it. Why are we modifying the EditableSkillDataHandler in the first place? I think you should probably leave it untouched.

What's the use case where the frontend needs to edit more than one skill simultaneously?

Reply:
`EditableSkillBackendApiService.fetchMultiSkills` line 247 of questions-list.directive

Reply:
It was used to get misconceptions from the skill backend dicts, but then I figured users don't need editable access to be able to get misconceptions, so I created a new handler - SkillDataHandler just now for that use.